nb-api/src/main/java/io/nosqlbench/nb/api/content/URIContent.java

```diff
@@ -0,0 +1,35 @@
+package io.nosqlbench.nb.api.content;
+
+import java.io.InputStream;
+import java.net.URI;
+import java.util.Objects;
+
+public class URIContent extends StreamContent<URI> {
```

Comment:
After some research, it appears that we can simply implement a https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/net/spi/URLStreamHandlerProvider.html and then have only URLs here. Are you interested in prototyping this?

Reply:
In other words, we can keep closer to the meaning of URLs and also avoid having two version of this concept here. It also would be a good illustration for how to add these for other protocols in the Java-prescribed way.

Reply:
Will absolutely take a look into the URLStreamHandlerProvider